src/ClientData/ModuleAndFunctionLookUp.cpp

```diff
@@ -0,0 +1,171 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "ClientData/ModuleAndFunctionLookUp.h"
+
+#include <ClientData/CaptureData.h>
+#include <ClientData/FunctionUtils.h>
+#include <ObjectUtils/Address.h>
+
+using orbit_client_protos::FunctionInfo;
+using orbit_client_protos::LinuxAddressInfo;
+
+namespace orbit_client_data {
+namespace {
+[[nodiscard]] std::optional<uint64_t>
+FindFunctionAbsoluteAddressByInstructionAbsoluteAddressUsingModulesInMemory(
+    const ProcessData* process, const ModuleManager* module_manager, uint64_t absolute_address) {
```

Comment:
Many of these functions take pointers to const, but assume the pointers are not `nullptr`. I think we should pass const references instead.